fedimint-client/src/client.rs

```diff
@@ -146,6 +147,18 @@ pub struct Client {
     request_hook: ApiRequestHook,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct FederationPreview {
+    config: ClientConfig,
```

Comment:
This should probably be `JsonClientConfig`, otherwise the module configs will be binary blobs